app/models/signbank/sign.rb

```diff
@@ -0,0 +1,72 @@
+module Signbank
+  class Sign < Record
+    self.table_name = :words
+    self.primary_key = :id
+
+    ##
+    # Exclude signs that either have the 'obscene' usage flag, or belong to the
+    # topic 'Sex and sexuality'. This filter is used in places where we feature
+    # signs the 'random sign' feature, and 'sign of the day', where it could be
+    # inappropriate to show such signs. We use HAVING here with a simple scoring
+    # algorithm, since SQLite does not support the EVERY operator. HAVING
+    # operates on a set of rows grouped by the word ID. If any of the rows in
+    # the groups contains the topic 'Sex and sexuality', we add 1 to the sum,
+    # otherwise we add 99. If the resulting number isn't divisible by 99, we
+    # know that at least one row contained 'Sex and Sexuality', and so should be
+    # excluded.
+    def self.safe_for_work
+      left_outer_joins(:topics)
+        .where("usage IS NULL OR usage != 'obscene'")
+        .group(:id)
+        .having <<~SQL.squish
+          SUM(
+            CASE WHEN word_topics.topic_name IS NOT NULL
+                 AND word_topics.topic_name = 'Sex and sexuality'
+            THEN 1 ELSE 99 END
+          ) % 99 = 0
```

Comment:
I mean... I would have thought there'd be a more conventional way to do this but as your SQL is general more advanced than mine, I'm assuming this is the best you were able to come up with?

Reply:
@mischa-s SQLite doesn't support the EVERY operator, this was the cleanest way I found to do this. I realise it's not the cleanest, which is why I added so much commentary around it.

Reply:
Specifically, this fixes an issue that otherwise comes up when a sign has more than one topic, but only one topic is 'Sex and sexuality'. Without this clause, the sign is still included as a candidate, because the LEFT OUTER JOIN (which is necessary to include signs that do not have any topics) includes a result row for each topic the sign has. 

So if a sign has the topics 'Animals' and 'Sex and sexuality', when I tried to do this with a simple WHERE clause, it was still being included in the result set - the row for the sign with the topic 'Sex and sexuality' was excluded, but the row for the sign with the topic 'Animals' was still there, so the sign was still included.

Having said this, I'll try and revisit this since I'm doing some other SQL improvement work today. Perhaps I can find a way to do this with a subselect instead.